src/core/opamProcess.ml

```diff
@@ -310,6 +310,13 @@ let set_resolve_command =
     resolve_command_fn := resolve_command
 let resolve_command cmd = !resolve_command_fn cmd
 
+let create_process_env cmd args env cin cout cerr =
+  let resolve cmd = OpamStd.Option.default cmd @@ resolve_command cmd in
+  if Sys.win32 && OpamStd.Sys.is_cygwin_variant (resolve cmd) = `Cygwin then
+    cygwin_create_process_env cmd args env cin cout cerr
+  else
+    Unix.create_process_env cmd args env cin cout cerr
```

Comment:
FWIW:
```suggestion
let create_process_env =
  if Sys.win32 then
    fun cmd ->
      let resolved_cmd = resolve_command cmd in
      if OpamStd.(Option.map_default Sys.is_cygwin_variant `Native resolved_cmd) = `Cygwin then
        cygwin_create_process_env cmd
      else
        Unix.create_process_env
  else
    Unix.create_process_env
```
The `Sys.win32` hoist is a personal meta-programming foible of mine (i.e. everything based on `Sys.win32` should be done once, as the value can't change!) but getting partial application `OpamProcess.create_process_env "foo"` to do the resolve immediately is useful/better.